ios/objective-c/v2/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleApp/Ampli/Ampli.m

```diff
@@ -0,0 +1,811 @@
+/**
+ * Ampli - A strong typed wrapper for your Analytics
+ *
+ * This file is generated by Amplitude.
+ * To update run 'ampli pull obj-c-ampli-v2'
+ *
+ * Required dependencies: AmplitudeSwift ~> 0.7.1
+ * Tracking Plan Version: 1
+ * Build: 1.0.0
+ * Runtime: ios:obj-c-ampli-v2
+ *
+ * [View Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest)
+ *
+ * [Full Setup Instructions](https://data.amplitude.com/test-codegen/Test%20Codegen/implementation/obj-c-ampli-v2)
+ */
+
+#import <Foundation/Foundation.h>
+@import AmplitudeSwift;
+#import "Ampli.h"
+
+@implementation Event: NSObject
```

Comment:
Can we make this extend `BaseEvent` rather than create a new class? Similar to Kotlin, TS, etc.

I would like to make it easy to use Ampli/Amplitude SDKs interchangably e.g. `amplitude.track(AmpliEvent(prop = true))`

Same for Swift codegen.

Reply:
Ok, I'll try. For Swift it can introduce some breaking changes.

Reply:
Thanks @falconandy. We are still in beta, so it's ok if you need breaking changes in Swift.

Reply:
Swift: done.
Objective-C: Subclassing Swift classes (`AMPBaseEvent`) from Objective-C is not supported (https://forums.swift.org/t/cannot-subclass-a-class-with-objc-subclassing-restricted-attribute/5674)

Reply:
Thanks @falconandy for ObjC could we write [ObjCBaseEvent.swift](https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/ObjC/ObjCBaseEvent.swift) in ObjC to work around this issue?

Reply:
@justin-fiedler I changed codegen - https://github.com/amplitude/ampli-examples/commit/77d9e041cc7cc41a06daf0c6151e98217e969caf
WDYT?

Reply:
Thanks @falconandy.

1. I think this works ok but would make setting schema based properties hard after initial creation via the static builder.

```
AMPBaseEvent *event = [EventWithAllProperties requiredBool:true ....]
[event.eventProperties.set:@"requiredBool" value: false]
```

2. Another option could be to keep the previous Ampli `Event` class and add a new `asBaseEvent(): AMPBaseEvent` but that also seems clunky and would like to avoid the extra type in the generated code.

3. Since we are declaring the new class ObjCBaseEvent class specific for ObjC usage I think that we could port ObjCBaseEvent implementation to be in Obj-C and use `NS_SWIFT` and extensions (if needed). See Itly SDK example here
* https://github.com/amplitude/itly-sdk-ios/tree/master/Sdk/Sdk

Could you please try to see if option 3 could work? Thanks

Reply:
@justin-fiedler About option 1:
Could you please clarify more - I think, nothing was changed here in comparison with Objective-C V1?

Reply:
@falconandy good point. I was thinking of TS which allows for easy setting of the generated event properties. I agree this provides the same functionality as Kotlin, Swift.

I think in this case the current implementation (option 1) is good. Will review again, but this this solution should work. Thanks